### ISSUE TITLE
Reindex broken in tf docker image (v1.11.0) -> Broken numpy dependency!

### ISSUE THREAD

**Issue by boldt** [tensorflow/tensorflow#22931]
# TLDR

The pandas, which is distributed with the tf docker image is broken!

# Description

I'm new to tf and I'm walking thru the [Machine Learning Crash Course](https://developers.google.com/machine-learning/crash-course/ml-intro) at the moment. [First Steps with TF](https://developers.google.com/machine-learning/crash-course/first-steps-with-tensorflow/programming-exercises) point me to the [Quick Introduction to pandas](https://colab.research.google.com/notebooks/mlcc/intro_to_pandas.ipynb?utm_source=mlcc&utm_campaign=colab-external&utm_medium=referral&utm_content=pandas-colab&hl=de) notebook. 

## Run example on colab

 Based on this. I created the following MWE on colab (actually, it is only pandas code):

https://colab.research.google.com/drive/19uDE_H4AtpLaEL6INrRrDMXkdANsNr69

It basically just does the following: 

1) Import the california housing data set
2) reorder the data set
3) prints the merged dataset

If you run my notebook, you can see that step 3 prints up to 20 different values for the data set (makred with `left only` and `right only`):

```
`   longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0      -114.3      34.2                15.0       5612.0          1283.0   
1      -114.5      34.4                19.0       7650.0          1901.0   
2      -114.6      33.7                17.0        720.0           174.0   
3      -114.6      33.6                14.0       1501.0           337.0   
4      -114.6      33.6                20.0       1454.0           326.0   
..        ...       ...                 ...          ...             ...   
15     -117.3      33.2                13.0       3619.0           791.0   
16     -118.3      33.8                25.0       4177.0           832.0   
17     -117.7      34.0                25.0       1859.0           463.0   
18     -118.1      34.0                50.0       1146.0           238.0   
19     -118.5      34.0                41.0       1240.0           320.0   

    population  households  median_income  median_house_value      _merge  
0       1015.0       472.0            1.5             66900.0   left_only  
1       1129.0       463.0            1.8             80100.0   left_only  
2        333.0       117.0            1.7             85700.0   left_only  
3        515.0       226.0            3.2             73400.0   left_only  
4        624.0       262.0            1.9             65500.0   left_only  
..         ...         ...            ...                 ...         ...  
15      1759.0       806.0            2.8             98500.0  right_only  
16      2123.0       789.0            5.1            446800.0  right_only  
17      1070.0       374.0            2.5            187500.0  right_only  
18       579.0       213.0            3.0            172600.0  right_only  
19       711.0       304.0            3.3            318100.0  right_only  
```

## Run example on local python

In have Python 3.6.6 and I install tf with `pip install  tensorflow`. 

Now I run the same code in a python script as follows (on Windows 10):

```
python Reorder.py
```

The reordered data is exactly the same as the imported data (all are marked with `both`):

```
longitude  latitude  housing_median_age  total_rooms  total_bedrooms  population  households  median_income  median_house_value _merge
0     -114.3      34.2                15.0       5612.0          1283.0      1015.0       472.0            1.5             66900.0   both
1     -114.5      34.4                19.0       7650.0          1901.0      1129.0       463.0            1.8             80100.0   both
2     -114.6      33.7                17.0        720.0           174.0       333.0       117.0            1.7             85700.0   both
3     -114.6      33.6                14.0       1501.0           337.0       515.0       226.0            3.2             73400.0   both
4     -114.6      33.6                20.0       1454.0           326.0       624.0       262.0            1.9             65500.0   both
5     -114.6      33.6                29.0       1387.0           236.0       671.0       239.0            3.3             74000.0   both
6     -114.6      33.6                25.0       2907.0           680.0      1841.0       633.0            2.7             82400.0   both
7     -114.6      34.8                41.0        812.0           168.0       375.0       158.0            1.7             48500.0   both
8     -114.6      33.6                34.0       4789.0          1175.0      3134.0      1056.0            2.2             58400.0   both
9     -114.6      34.8                46.0       1497.0           309.0       787.0       271.0            2.2             48100.0   both
```

## Run example in tf docker container

Now I run the same python script in the [tf docker container](https://hub.docker.com/r/tensorflow/tensorflow/) (v1.11.0) as follows (on Windows 10):

```bash
docker run --rm -it -v C:\Users\boldt\docker\tf\scripts\:/scripts tensorflow/tensorflow:1.11.0-py3 python /scripts/Reorder.py
```

The result is the same. The reordered data is exactly the same as the imported data (all are marked with `both`):

# Conclusion

Because of this, I am unable to randomize the data set and split it in a training and a test set using the tf docker container. Currently, my more complex example always results the same training and test sets.

**Comment by boldt** [comment]
Since my MWE is reduced to only panda code without tf dependencies, I assumed an issue in pandas:

* colab uses pandas ``0.22.0`
* tf docker container (v1.11.0) uses pandas ``0.23.4``
* my local python installation uses pandas ``0.23.4``

I downgraded pandas on my local environment now:

* 0.22.0 (`pip install --force-reinstall pandas==0.22.0`) - Reorder works
* 0.23.0 (`pip install --force-reinstall pandas==0.23.0`) - Broken

# Conclusion

The panda version >=0.23.0, which is distributed with the tf docker image is broken!

**Comment by harshini-gadige** [comment]
@boldt Hi, as there is a new ticket raised in Pandas/Numpy, feel free to close the issue here. Or else if you would like to add any other points on this, please do. Thank you !

**Comment by boldt** [comment]
[Numby v1.15.3 was released on Oct, 22nd](https://github.com/numpy/numpy/releases/tag/v1.15.3). I just checked the tag `nightly-py3`. It contains the newest numpy version. 

I'll close this ticket as soon the new docker image 1.12.0 is released (rc1 is out and does not contain the newest numpy yet)

**Comment by harshini-gadige** [comment]
> Just saw, that 1.12.0 is pushes to docker hub. Thus I'm closing it.

This is still open :)

**Comment by tensorflowbutler** [comment]
Nagging Assignee @harshini-gadige: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by boldt** [comment]
> This is still open :)

Thanks
